client/src/connection_cache.rs

```diff
@@ -54,8 +56,12 @@ fn get_connection(addr: &SocketAddr) -> Connection {
     let mut map = (*CONNECTION_MAP).lock().unwrap();
 
     match map.map.get(addr) {
-        Some(connection) => connection.clone(),
+        Some(connection) => {
+            inc_new_counter_info!("get_connection-cache-hits", 1);
```

Comment:
prefer a metrics struct that's submitted periodically as a single datapoint.

`inc_new_counter_*` are awful for actually examining as they're submitted as independent measurements, rather than fields under the same measurement.  this makes them very difficult to compare